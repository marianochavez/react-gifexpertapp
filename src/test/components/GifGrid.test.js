import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");


describe("Test on <GifGrid/>", () => {

    const category = "Naruto";


    test('should show GifGrid correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('should show items when use useFetchGifs', () => {
        
        const gifs = [
            {
                id: "ABC",
                url: "https://localhost/",
                title: "Naruto"
            },
            {
                id: "DEF",
                url: "https://localhost/",
                title: "Naruto"
            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false);
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
    })
    
    
});


