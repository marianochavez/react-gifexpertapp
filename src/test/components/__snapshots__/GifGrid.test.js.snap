// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifGrid/> should show GifGrid correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Naruto
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test on <GifGrid/> should show items when use useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Naruto
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Naruto"
      url="https://localhost/"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="Naruto"
      url="https://localhost/"
    />
  </div>
</Fragment>
`;
